src/pymor/algorithms/newton.py

```diff
@@ -102,60 +110,103 @@ def newton(operator, rhs, initial_guess=None, mu=None, error_product=None, least
+            update = jacobian.apply_inverse(residual, least_squares=least_squares)
         except InversionError:
             raise NewtonError('Could not invert jacobian')
+
+        # compute step size
         if isinstance(relax, Number):
             step_size = relax
         elif relax == 'armijo':
-            logger.info(f'Using Armijo as line search method')
             def res(x):
                 residual_vec = rhs - operator.apply(x, mu=mu)
                 return residual_vec.norm(error_product)[0]
+
             if error_product is None:
                 grad = - (jacobian.apply(residual) + jacobian.apply_adjoint(residual))
             else:
-                grad = - (jacobian.apply_adjoint(error_product.apply(residual)) + jacobian.apply(error_product.apply_adjoint(residual)))
-            step_size = armijo(res, U, correction, grad=grad, initial_value=err, **(line_search_params or {}))
+                grad = - (jacobian.apply_adjoint(error_product.apply(residual))
+                          + jacobian.apply(error_product.apply_adjoint(residual)))
+            step_size = armijo(res, U, update, grad=grad, initial_value=residual_norm, **(line_search_params or {}))
         else:
             raise ValueError('Unknown line search method')
-        U.axpy(step_size, correction)
+
+        # update solution and residual
+        U.axpy(step_size, update)
         residual = rhs - operator.apply(U, mu=mu)
 
-        err = residual.norm(error_product)[0]
-        logger.info(f'Iteration {iteration:2}: Residual: {err:5e},  '
-                    f'Reduction: {err / error_sequence[-1]:5e}, Total Reduction: {err / error_sequence[0]:5e}')
-        error_sequence.append(err)
-        if not np.isfinite(err):
+        # compute norms
+        solution_norm = U.norm(error_product)[0]
+        solution_norms.append(solution_norm)
+        update_norm = update.norm(error_product)[0] * step_size
+        update_norms.append(update_norm)
+        residual_norm = residual.norm(error_product)[0]
+        residual_norms.append(residual_norm)
+
+        # select error measure for next iteration
+        err = residual_norm if error_measure == 'residual' else update_norm
+        if error_measure == 'update':
+            err_scale_factor = solution_norm
+
+        logger.info(f'it:{iteration} '
+                    f'res:{residual_norm:.3e} '
+                    f'red:{residual_norm / residual_norms[-2]:.3e} '
+                    f'tot_red:{residual_norm / residual_norms[0]:.3e} '
+                    f'norm:{solution_norm:.3e} '
+                    f'upd:{update_norm:.3e} '
+                    f'rel_upd:{update_norm / solution_norm:.3e}')
+
+        if not np.isfinite(residual_norm) or not np.isfinite(solution_norm):
             raise NewtonError('Failed to converge')
 
-    data['error_sequence'] = np.array(error_sequence)
+    logger.info('')
+
+
+    data['solution_norms'] = np.array(solution_norms)
```

Comment:
Just for my understanding: which cases do you have in mind where the user is interested in the norms of intermediate solutions of the Newton algorithm? The norms of the updates as well as the residuals could be of great interest, that's right, but I don't see that for the norms of the solutions.

Reply:
I have no concrete use cases in mind. However, the norms are computed anyway, so it doesn't hurt to return them